### ISSUE TITLE
Buggy processing of column named "row"

### ISSUE THREAD

**Issue by pyhedgehog** [saltcorn/saltcorn#1801]
**Describe the bug**
I've got error running my view:
```
In some_list view (List viewtemplate):
Duplicate parameter name not allowed in this context.
```

**To Reproduce**
1. Create table with column named "row" (I've used it for JSON, but it seems not important)
2. Create view (almost any type) with formula value.
3. Run it.

**Expected behavior**
Options:
1. View runs correctly.
2. Creation of such column prohibited.

**Core reason**
https://github.com/saltcorn/saltcorn/blob/011e8cbc086a6176baf74904befccd89f9f95381/packages/saltcorn-data/models/expression.ts#L508-L510
"user" field checked, but "row" not checked.
I propose something like:
```javascript
function eval_expression(expression: string, row: any, user?: any): any {
  const field_names = Object.keys(row);
  const dummy_prefix = `_${field_names.length}_`;
  const user_arg = `${field_names.includes("user")?dummy_prefix:''}user`;
  const row_arg = `${field_names.includes("row")?(field_names.includes("table_row")?dummy_prefix:'table_'):''}row`;
  const args = [row_arg, `{${field_names.join()}}`, user_arg].join(', ');
  const { getState } = require("../db/state");
  return runInNewContext(
    `(${args})=>(${expression})`,
    getState().function_context
  )(row, row, user);
}
```
However this solution is questionable.

**Comment by glutamate** [comment]
For similar reasons "function" is not a good column name

**Comment by pyhedgehog** [comment]
Any JS keyword (`if`, `for`, `case` and so on) will result in error here, but "user" and "row" aren't JS or PG keywords.
This means that in addition to my proposed change there should be list of keywords that checked on column creation/renaming and at least warning, like "this column name belongs to everincreasing list of keywords and cause unpredictable errors".

**Comment by AleksandrSokolov** [comment]
@glutamate @pyhedgehog What we decide to do with this issue?

**Comment by glutamate** [comment]
fixed in #1808 (https://github.com/saltcorn/saltcorn/pull/1808/commits/b0e2b5f503d88b2849ff609519f8b7d3724103fe)
